osu.Game/Graphics/Containers/TapToConfirmContainer.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Graphics.Containers
+{
+    public class TapToConfirmContainer : Container
+    {
+        public Action Action;
+
+        /// <summary>
+        /// Whether currently in a fired state (and the confirm <see cref="Action"/> has been sent).
+        /// </summary>
+        private bool fired { get; set; }
+
+        protected void BeginConfirm()
+        {
+            if (fired) return;
+
+            Action?.Invoke();
+            fired = true;
+        }
```

Comment:
This is really weird naming and structure. Not really sure what it's trying to achieve (all the actions you've implemented would guarantee they are only run once, I think).